elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+import lxml.html
+import random
+import chevron
+import base64
+import os
+
+# Read https://prairielearn.readthedocs.io/en/latest/devElements/
+# Official documentation on making custom PL
+
+
+def render_html_colour(score):
+    # used to render the correct colour depending on student score
+    if score == 0:
+        return 'badge-danger'
+    elif score == 1.0:
+        return 'badge-success'
+    else:
+        return 'badge-warning'
+
+# courtesy of https://github.com/PrairieLearn/PrairieLearn/pull/2625
+
+
+def getallAnswer(submitted_blocks, block_indents, leading_code, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer + trailing_code + '\n'
+    return answer
+
+
+def gettarilAnswer(submitted_blocks, block_indents, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = answer + trailing_code + '\n'
+    return answer
+
+
+def getleadAnswer(submitted_blocks, block_indents, leading_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer_indent in enumerate(block_indents):
+        indent = int(answer_indent)
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer
+    return answer
+
+
+def render(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    if data['panel'] == 'question':
+        mcq_options = []   # stores MCQ options
+        student_previous_submission = []
+        submission_indent = []
+
+        pl.check_attribs(element,
+                         required_attribs=['answers-name'],
+                         optional_attribs=['shuffle-options',
+                                           'permutation-mode',
+                                           'check-indentation',
+                                           'header-left-column',
+                                           'header-right-column',
+                                           'external-grader',
+                                           'file-name',
+                                           'leading-code',
+                                           'trailing-code',
+                                           'max-distractors',   # Legacy attribute
+                                           'max-feedback-count'])  # Legacy attribute
+
+        for html_tags in element:
+            if html_tags.tag == 'pl-answer':
+                # CORRECT is optional for backward compatibility
```

Comment:
I thought we decided last meeting that we shouldn't be doing anything for backwards compatibility?

I think any references to "backwards compatibility" in this code should be reviewed.

Reply:
yeah, this was what I added before the meeting to make legacy questions work, I could remove them